drivers/adc/adc_stm32.c

```diff
@@ -496,6 +505,9 @@ static void adc_stm32_calibration_start(const struct device *dev)
 	defined(CONFIG_SOC_SERIES_STM32WBX) || \
 	defined(CONFIG_SOC_SERIES_STM32G4X)
 	LL_ADC_StartCalibration(adc, LL_ADC_SINGLE_ENDED);
+	if (config->differential_channels_used) {
+		LL_ADC_StartCalibration(adc, LL_ADC_DIFFERENTIAL_ENDED);
```

Comment:
mmmh is it really ok to immediately start a calibration without waiting for the previous one to be completed? (checking `if LL_ADC_IsCalibrationOnGoing(...)`)

Reply:
You're right, I missed that point.
I think the most simple solution is to add a parameter `bool single_ended` to adc_stm32_calibration_start, and check it. If true, then we do single-ended calibration, otherwise differential. And it should be marked as unused for series that do not support differential mode to avoid any warnings.